core/camel-core-model/src/main/java/org/apache/camel/model/PausableDefinition.java

```diff
@@ -0,0 +1,114 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.camel.model;
+
+import java.util.function.Predicate;
+
+import javax.xml.bind.annotation.XmlAccessType;
+import javax.xml.bind.annotation.XmlAccessorType;
+import javax.xml.bind.annotation.XmlAttribute;
+import javax.xml.bind.annotation.XmlRootElement;
+import javax.xml.bind.annotation.XmlTransient;
+
+import org.apache.camel.ConsumerListener;
+import org.apache.camel.spi.Metadata;
+
+/**
+ * Resume EIP to support resuming processing from last known offset.
+ */
+@Metadata(label = "eip,routing")
+@XmlRootElement(name = "pausable")
+@XmlAccessorType(XmlAccessType.FIELD)
+public class PausableDefinition extends NoOutputDefinition<PausableDefinition> {
+
+    @XmlTransient
+    private ConsumerListener<?, ?> consumerListenerBean;
+
+    @XmlAttribute(required = true)
+    @Metadata(required = true, javaType = "org.apache.camel.ConsumerListener")
+    private String consumerListener;
+
+    @XmlTransient
+    private Predicate<?> untilCheck;
+
+    @XmlAttribute(required = true)
+    private String untilCheckRef;
```

Comment:
Avoid using xxxRef style, but just name it xxx
Also set javaType = "...Predicate" as its a known class

Reply:
In this case it would clash with the untilCheck variable above, but I will name it something else.

Reply:
Ah, I'll do like the resume api definition and rename the predicate instead ;)